src/EnergyPlus/VariableSpeedCoils.cc

```diff
@@ -5116,291 +5023,205 @@ namespace VariableSpeedCoils {
-             state.dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).VSCoilTypeOfNum == Coil_HeatingWaterToAirHPVSEquationFit)) {
+        if ((varSpeedCoils.RatedWaterVolFlowRate == AutoSize) && (varSpeedCoils.VSCoilTypeOfNum == Coil_CoolingWaterToAirHPVSEquationFit ||
+                                                                  varSpeedCoils.VSCoilTypeOfNum == Coil_HeatingWaterToAirHPVSEquationFit)) {
             RatedWaterFlowAutoSized = true;
         }
 
-        //!   if not found on a plant loop, check condenser loop and warn user if not found
-        //    IF(PltSizNum == 0) THEN
-        //      PltSizNum = &
-        //          MyCondPlantSizingIndex('COIL:'//TRIM(VarSpeedCoil(DXCoilNum)%CoolHeatType)// &
-        //               TRIM(CurrentObjSubfix), &
-        //                                 VarSpeedCoil(DXCoilNum)%Name, &
-        //                                 VarSpeedCoil(DXCoilNum)%WaterInletNodeNum, &
-        //                                 VarSpeedCoil(DXCoilNum)%WaterOutletNodeNum, ErrorsFound)
-        //    END IF
-
         //   WSHP condenser can be on either a plant loop or condenser loop. Test each to find plant sizing number.
         //   first check to see if coil is connected to a plant loop, no warning on this CALL
         if (RatedWaterFlowAutoSized) {
-            if (state.dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).CondenserType == DataHeatBalance::RefrigCondenserType::Water)
+            if (varSpeedCoils.CondenserType == DataHeatBalance::RefrigCondenserType::Water)
                 PltSizNum = MyPlantSizingIndex(state,
-                                               "COIL:" + state.dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).CoolHeatType + CurrentObjSubfix,
-                                               state.dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).Name,
-                                               state.dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).WaterInletNodeNum,
-                                               state.dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).WaterOutletNodeNum,
+                                               "COIL:" + varSpeedCoils.CoolHeatType + CurrentObjSubfix,
+                                               varSpeedCoils.Name,
+                                               varSpeedCoils.WaterInletNodeNum,
+                                               varSpeedCoils.WaterOutletNodeNum,
                                                ErrorsFound,
                                                false);
 
             if (PltSizNum > 0) {
                 rho = GetDensityGlycol(state,
-                                       state.dataPlnt->PlantLoop(state.dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).plantLoc.loopNum).FluidName,
+                                       state.dataPlnt->PlantLoop(varSpeedCoils.plantLoc.loopNum).FluidName,
```

Comment:
Looking up fluid properties using fluid name pisseth me off.

Reply:
I understand that. That was clear from when @jmythms cleaned up FluidProperties to use `std::array`. Regardless, this is a stupid idiom. In general, having functions that do one thing on the first call and something else on subsequent calls is kind of dumb. Because you have to pass the parameters for both cases every time and you also have to check which case you are in every time. A better idiom is what you suggested. Have one function that is only called for setup during initialization, and then a separate function that is called the rest of the time that does not take the initialization parameters or do the initialization check. This is a general statement that applies to many things in EnergyPlus. I was just venting here. It's been a week/month/year.